app/services/update_task_service.rb

```diff
@@ -0,0 +1,62 @@
+class UpdateTaskService
+  class << self
+    def perform(task, user, task_params)
+      task = handle_status_update(task, user, task_params)
+      task = handle_employee_assignment(task, user, task_params) unless task.errors.any?
+      task = handle_other_changes(task, user, task_params) unless task.errors.any?
+
+      task
+    end
+
+    def handle_other_changes(task, user, task_params)
+      params = task_params.except(:assigned_employee_ids, :status)
+      task.update(params)
+      task
+    end
+
+    def handle_employee_assignment(task, user, task_params)
```

Comment:
This method is a bit complex and could probably be streamlined given a bit more thinking time on it (including not just readability but also performance)